### PR TITLE
[nibeheatpump] Removed some unneeded isTraceEnabled and use the ESH HexUtils

### PR DESCRIPTION
**Pull request by martinvw**
Follow up of #3445

### CODE REVIEW

**Comment by wborn** [comment]
Can you also replace the other occurences of `DatatypeConverter.parseHexBinary` in this binding @martinvw? :-)

It is also extensively used in org.openhab.binding.nibeheatpump.test.

**Comment by martinvw** [comment]
I already saw that coming when I touched the first one 😁

**Comment by martinvw** [comment]
Okay, its updated, feel free to re-review @wborn it includes some more changes now :-)

File: addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SimulatorConnector.java [hunk 1]
```diff
@@ -39,13 +39,13 @@
...
-    private List<byte[]> readQueue = new ArrayList<byte[]>();
-    private List<byte[]> writeQueue = new ArrayList<byte[]>();
+    private final List<byte[]> readQueue = new ArrayList<>();
+    private final List<byte[]> writeQueue = new ArrayList<>();
 
     private static final Random RANDOM = new Random();
 
     @SuppressWarnings("serial")
-    ArrayList<ModbusValue> dataReadoutValues = new ArrayList<ModbusValue>() {
+    final ArrayList<ModbusValue> dataReadoutValues = new ArrayList<ModbusValue>() {
```
[the remark this answers is not in the document]
Comment (martinvw) [reply addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SimulatorConnector.java]: Nope it’s an anonymous class

File: addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SimulatorConnector.java [hunk 2]
```diff
@@ -70,14 +70,14 @@
         }
     };
 
-    private Map<Integer, Integer> cache = Collections.synchronizedMap(new HashMap<Integer, Integer>());
+    private final Map<Integer, Integer> cache = Collections.synchronizedMap(new HashMap<Integer, Integer>());
```
[the remark this answers is not in the document]
Comment (martinvw) [reply addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SimulatorConnector.java]: I fear not but let’s see that in my ide after my son empties his bottle of milk 🥛😉

**Comment by martinvw** [comment]
@kaikreuzer / @paulianttila your opinion is appreciated

@kaikreuzer yours mainly about the static logger from a static context, but I believe we agreed upon this before, but just as a sanity check.

**Comment by martinvw** [comment]
Thanks all for checking!

**Pull request merged**